sys/include/net/sock/dtls.h

```diff
@@ -657,28 +669,27 @@ ssize_t sock_dtls_recv_buf(sock_dtls_t *sock, sock_dtls_session_t *remote,
  *                      if no session exist between client and server.
  * @param[in] data      Pointer where the data to be send are stored
  * @param[in] len       Length of @p data to be send
- *
- * @note Function may block until a session is created if there is no
- *       existing session with @p remote.
- *
- * @note Initiating a session through this function will require
- * @ref sock_dtls_recv() called from another thread to receive the handshake
- * messages.
+ * @param[in] timeout   Handshake timeout in microseconds.
+ *                      If `0`, the function returns after new handshake process
+ *                      is started.
```

Comment:
I am not sure that is a good idea, API-wise. IMHO, with timeout=0 the session should already be created. If not, the function should just return an error and not inititialize anything.

Reply:
I prefer to stick with the current behavior for now as that is how send in tinydtls and WolfSSL is done but I can see how the current docs can also be confusing. Rewrote the description of the `timeout` parameter.

Reply:
Mhhhh... but doesn't this introduce some form of implicit synchronization? I just fear, that we would be impeded in #12907 again.

Reply:
yeah that is valid concern. `sock_dtls_send` now does not start a new handshake if `timeout == 0`.